src/pages/index.js

```diff
@@ -0,0 +1,293 @@
+    if (openedModal) closeModal(openedModal);
+  }
+}
+
+// Overlay click close handler
+function handleOverlayClick(event) {
+  if (event.target.classList.contains("modal_opened")) {
+    closeModal(event.target);
+  }
+}
+
+// Helper to set up modal close buttons
+function setUpCloseButtons() {
+  document.querySelectorAll(".modal__close-btn").forEach((button) => {
+    const modal = button.closest(".modal");
+    button.addEventListener("click", () => closeModal(modal));
+  });
+}
+setUpCloseButtons();
+
+// Load user info and cards
+api
+  .getAppInfo()
+  .then(([cards, userInfo]) => {
+    document.querySelector(".profile__avatar").src = userInfo.avatar;
+    document.querySelector(".profile__name").textContent = userInfo.name;
+    document.querySelector(".profile__description").textContent =
+      userInfo.about;
+
+    const cardsList = document.querySelector(".cards__list");
+    cards
+      .reverse()
+      .forEach((cardData) => cardsList.prepend(createCardElement(cardData)));
+  })
+  .catch(console.error);
+
+// Card creation function
+function createCardElement(data) {
+  const template = document.querySelector("#card-template");
+  const cardElement = template.content.querySelector(".card").cloneNode(true);
+
+  const cardImageEl = cardElement.querySelector(".card__image");
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardElement.querySelector(".card__title").textContent = data.name;
+
+  setUpImagePreview(cardImageEl, data);
+  setUpLikeButton(cardElement.querySelector(".card__like-btn"), data);
+  cardElement
+    .querySelector(".card__delete-btn")
+    .addEventListener("click", () => openDeleteModal(cardElement, data._id));
+
+  return cardElement;
+}
+
+// Helper function to set up image preview
+function setUpImagePreview(imageElement, data) {
+  imageElement.addEventListener("click", () => {
+    const modalImage = previewModal.querySelector(".modal__image");
+    const modalCaption = previewModal.querySelector(".modal__caption");
+
+    modalImage.src = data.link;
+    modalImage.alt = data.name;
+    modalCaption.textContent = data.name;
+
+    openModal(previewModal);
+  });
+}
+
+// Function for Like button handler
+function setUpLikeButton(likeButton, data) {
+  if (data.isLiked) likeButton.classList.add("card__like-btn_liked");
+
+  likeButton.addEventListener("click", () => {
+    const isLiked = likeButton.classList.contains("card__like-btn_liked");
+    api
+      .handleLikeStatus(data._id, isLiked)
+      .then((updatedData) =>
+        likeButton.classList.toggle("card__like-btn_liked", updatedData.isLiked)
+      )
+      .catch(console.error);
+  });
+}
+
+// Function for Delete modal logic
+function openDeleteModal(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+
+  const deleteButton = deleteForm.querySelector(".modal__submit-btn");
+  showLoading(deleteButton, false, "Deleting..", "Delete"); // Reset button text to "Delete"
```

Comment:
This is not needed here. You open the confirmation modal and set `selectedCard` and `selectedCardId` (that's it). You don't need to set a submit handler on opening.


The delete form submit handler should be added only **once** because `selectedCard` and `selectedCardId` will be accessible inside it

Only these 3 lines should be here

```js

function openDeleteModal(cardElement, cardId) {
  selectedCard = cardElement;
  selectedCardId = cardId;
  openModal(deleteModal);
}

```